### ISSUE TITLE
Error viewing cookbook file

### ISSUE THREAD

**Issue by joerg** [3ofcoins/chef-browser#59]
I am playing around with your Docker containers for chef server and this chef browser and I do get a strange error when trying to access any cookbook file. Viewing any other information (environments, nodes, data bags, cookbook info) works just fine.

This is the error page I get (sorry for the bad formatting, just a copy/paste of the full page).

```
application error
MentosError at /cookbook/rcrs-base-0.1.25/recipes/certificates.rb
Failed to get header.

    file: popen.rb location: rescue in get_header line: 381 

BACKTRACE

(condense)

JUMP TO: GET POST COOKIES ENV

    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/pygments.rb-0.6.3/lib/pygments/popen.rb in rescue in get_header
                raise MentosError, "Failed to get header."
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/pygments.rb-0.6.3/lib/pygments/popen.rb in get_header
              begin
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/pygments.rb-0.6.3/lib/pygments/popen.rb in block in mentos
                  header = get_header
    /usr/lib/ruby/2.2.0/timeout.rb in block in timeout
                return yield(sec)
    /usr/lib/ruby/2.2.0/timeout.rb in block in catch
              ::Kernel.catch(exc) {yield exc}
    /usr/lib/ruby/2.2.0/timeout.rb in catch
              ::Kernel.catch(exc) {yield exc}
    /usr/lib/ruby/2.2.0/timeout.rb in catch
              ::Kernel.catch(exc) {yield exc}
    /usr/lib/ruby/2.2.0/timeout.rb in timeout
              bt = Error.catch(message, &bl)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/pygments.rb-0.6.3/lib/pygments/popen.rb in mentos
                Timeout::timeout(timeout_time) do
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/pygments.rb-0.6.3/lib/pygments/popen.rb in highlight
              str = mentos(:highlight, nil, opts, code)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/pygments.rb-0.6.3/lib/pygments/lexer.rb in highlight
              Pygments.highlight(text, options)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/github-linguist-3.5.2/lib/linguist/language.rb in colorize
              lexer.highlight(text, options)
    /opt/chef-browser/lib/chef-browser/file_content.rb in highlight_file
                  lexer.colorize(content, options: HIGHLIGHT_OPTIONS)
    /opt/chef-browser/lib/chef-browser/file_content.rb in show_file
                    FileContent.highlight_file(content.name, extname, content.data)
    /opt/chef-browser/lib/chef-browser/app.rb in block in <class:App>
                content: FileContent.show_file(cookbook_file, uri_options)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in call
                  proc { |a,p| unbound_method.bind(a).call }
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in block in compile!
                  proc { |a,p| unbound_method.bind(a).call }
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in []
                    route_eval { block[*args] }
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in block (3 levels) in route!
                    route_eval { block[*args] }
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in route_eval
              throw :halt, yield
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in block (2 levels) in route!
                    route_eval { block[*args] }
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in block in process_route
                block ? block[self, values] : yield(self, values)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in catch
              catch(:pass) do
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in process_route
              catch(:pass) do
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in block in route!
                  returned_pass_block = process_route(pattern, keys, conditions) do |*args|
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in each
                routes.each do |pattern, keys, conditions, block|
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in route!
                routes.each do |pattern, keys, conditions, block|
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in block in dispatch!
                route!
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in block in invoke
              res = catch(:halt) { yield }
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in catch
              res = catch(:halt) { yield }
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in invoke
              res = catch(:halt) { yield }
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in dispatch!
              invoke do
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in block in call!
              invoke { dispatch! }
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in block in invoke
              res = catch(:halt) { yield }
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in catch
              res = catch(:halt) { yield }
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in invoke
              res = catch(:halt) { yield }
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in call!
              invoke { dispatch! }
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in call
              dup.call!(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/session/abstract/id.rb in context
                  status, headers, body = app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/session/abstract/id.rb in call
                  context(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-protection-1.5.3/lib/rack/protection/xss_header.rb in call
                status, headers, body = @app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-protection-1.5.3/lib/rack/protection/path_traversal.rb in call
                app.call env
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-protection-1.5.3/lib/rack/protection/json_csrf.rb in call
                status, headers, body = app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-protection-1.5.3/lib/rack/protection/base.rb in call
                result or app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-protection-1.5.3/lib/rack/protection/base.rb in call
                result or app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-protection-1.5.3/lib/rack/protection/frame_options.rb in call
                status, headers, body        = @app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/nulllogger.rb in call
              @app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/head.rb in call
            status, headers, body = @app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/show_exceptions.rb in call
              @app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in call
              result, callback = app.call(env), env['async.callback']
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in call
              @stack.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/tempfile_reaper.rb in call
              status, headers, body = @app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/lint.rb in _call
              status, headers, @body = @app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/lint.rb in call
              dup._call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/showexceptions.rb in call
              @app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/commonlogger.rb in call
              status, header, body = @app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb in call
                call_without_check(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/chunked.rb in call
              status, headers, body = @app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/content_length.rb in call
              status, headers, body = @app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/puma-2.14.0/lib/puma/server.rb in handle_request
                  status, headers, res_body = @app.call(env)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/puma-2.14.0/lib/puma/server.rb in process_client
                  case handle_request(client, buffer)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/puma-2.14.0/lib/puma/server.rb in block in run
                    process_client client, buffer
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/puma-2.14.0/lib/puma/thread_pool.rb in call
                    block.call(work, *extra)
    /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/puma-2.14.0/lib/puma/thread_pool.rb in block in spawn_thread
                    block.call(work, *extra)

GET

No GET data.
POST

No POST data.
COOKIES

No cookie data.
Rack ENV
Variable    Value
GATEWAY_INTERFACE   
CGI/1.2
HTTP_ACCEPT     
text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
HTTP_ACCEPT_ENCODING    
gzip, deflate
HTTP_ACCEPT_LANGUAGE    
en-US,en;q=0.5
HTTP_CONNECTION     
keep-alive
HTTP_DNT    
1
HTTP_HOST   
127.0.0.1:9292
HTTP_REFERER    
http://127.0.0.1:9292/cookbook/rcrs-base-0.1.25?recipes
HTTP_USER_AGENT     
Mozilla/5.0 (Windows NT 6.1; WOW64; rv:42.0) Gecko/20100101 Firefox/42.0
HTTP_VERSION    
HTTP/1.1
PATH_INFO   
/cookbook/rcrs-base-0.1.25/recipes/certificates.rb
QUERY_STRING    
REMOTE_ADDR     
10.0.2.2
REQUEST_METHOD  
GET
REQUEST_PATH    
/cookbook/rcrs-base-0.1.25/recipes/certificates.rb
REQUEST_URI     
/cookbook/rcrs-base-0.1.25/recipes/certificates.rb
SCRIPT_NAME     
SERVER_NAME     
127.0.0.1
SERVER_PORT     
9292
SERVER_PROTOCOL     
HTTP/1.1
SERVER_SOFTWARE     
2.14.0
puma.socket     
#<TCPSocket:fd 75>
rack.after_reply    
[]
rack.errors     
#<Object:0x000000034a5710>
rack.hijack     
#<Proc:0x007f0afa2d20b0@/opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/lint.rb:525>
rack.hijack?    
true
rack.input  
#<Rack::Lint::InputWrapper:0x007f0afa2d1cc8 @input=#<Puma::NullIO:0x000000052b65d8>>
rack.logger     
#<Rack::NullLogger:0x000000050a65e0 @app=#<Rack::Protection::FrameOptions:0x000000050a66d0 @app=#<Rack::Protection::HttpOrigin:0x000000050a6770 @app=#<Rack::Protection::IPSpoofing:0x000000050a6810 @app=#<Rack::Protection::JsonCsrf:0x000000050a68b0 @app=#<Rack::Protection::PathTraversal:0x000000050a6950 @app=#<Rack::Protection::XSSHeader:0x000000050a6a40 @app=#<Rack::Session::Cookie:0x00000005063060 @secrets=["SPsYUwecQC401S/wBLgcNejabIV5Q5lJSu5DLPAjcP+BXPbroTMgvsAOCfUMZNg7CmtpveWEGWS6QxJG32MCdw=="], @coder=#<Rack::Session::Cookie::Base64::Marshal:0x00000005062fe8>, @app=#<ChefBrowser::App:0x000000050487d8 @default_layout=:layout, @preferred_extension=nil, @app=nil, @template_cache=#<Tilt::Cache:0x000000050487b0 @cache={}>>, @default_options={:path=>"/", :domain=>nil, :expire_after=>3600, :secure=>false, :httponly=>true, :defer=>false, :renew=>false, :sidbits=>128, :secure_random=>SecureRandom, :secret=>"SPsYUwecQC401S/wBLgcNejabIV5Q5lJSu5DLPAjcP+BXPbroTMgvsAOCfUMZNg7CmtpveWEGWS6QxJG32MCdw==", :coder=>#<Rack::Session::Cookie::Base64::Marshal:0x00000005062fe8>}, @key="rack.session", @cookie_only=true, @sidbits=128, @sid_secure=SecureRandom, @sid_length=32>, @options={:reaction=>:drop_session, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml"], :xss_mode=>:block, :nosniff=>true, :except=>[:session_hijacking, :remote_token]}>, @options={:reaction=>:drop_session, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml"], :except=>[:session_hijacking, :remote_token]}>, @options={:reaction=>:drop_session, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml"], :except=>[:session_hijacking, :remote_token]}>, @options={:reaction=>:drop_session, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml"], :except=>[:session_hijacking, :remote_token]}>, @options={:reaction=>:drop_session, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml"], :except=>[:session_hijacking, :remote_token]}>, @options={:reaction=>:drop_session, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml"], :frame_options=>:sameorigin, :except=>[:session_hijacking, :remote_token]}, @frame_options="SAMEORIGIN">>
rack.multiprocess   
false
rack.multithread    
true
rack.request.cookie_hash    
{}
rack.request.query_hash     
{}
rack.request.query_string   
rack.run_once   
false
rack.session    
#<Rack::Session::Abstract::SessionHash:0x3f857d168a90 not yet loaded>
rack.session.options    
{:path=>"/", :domain=>nil, :expire_after=>3600, :secure=>false, :httponly=>true, :defer=>false, :renew=>false, :sidbits=>128, :secure_random=>SecureRandom, :secret=>"SPsYUwecQC401S/wBLgcNejabIV5Q5lJSu5DLPAjcP+BXPbroTMgvsAOCfUMZNg7CmtpveWEGWS6QxJG32MCdw==", :coder=>#<Rack::Session::Cookie::Base64::Marshal:0x00000005062fe8>}
rack.tempfiles  
[]
rack.url_scheme     
http
rack.version    
[1, 3]
sinatra.accept  
[#<Sinatra::Request::AcceptEntry:0x00000004eb06a0 @entry="text/html", @type="text/html", @params={}, @q=1.0>, #<Sinatra::Request::AcceptEntry:0x00000004eb0010 @entry="application/xhtml+xml", @type="application/xhtml+xml", @params={}, @q=1.0>, #<Sinatra::Request::AcceptEntry:0x00000004ecb978 @entry="application/xml;q=0.9", @type="application/xml", @params={}, @q=0.9>, #<Sinatra::Request::AcceptEntry:0x00000004ecb0e0 @entry="*/*;q=0.8", @type="*/*", @params={}, @q=0.8>]
sinatra.commonlogger    
true
sinatra.error   
#<MentosError: Failed to get header.>
sinatra.route   
GET /cookbook/:cookbook/*

You're seeing this error because you have enabled the show_exceptions setting.
```

docker logs shows this:

```
10.0.2.2 - - [24/Nov/2015:14:33:43 +0000] "GET /js/jquery.filtertable.js HTTP/1.1" 200 7107 0.0045
10.0.2.2 - - [24/Nov/2015:14:33:43 +0000] "GET /js/jquery-2.0.3.js HTTP/1.1" 200 242142 0.0250
10.0.2.2 - - [24/Nov/2015:14:33:43 +0000] "GET /js/bootstrap.js HTTP/1.1" 200 58533 0.0072
2015-11-24 14:33:44 - MentosError - Failed to get header.:
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/pygments.rb-0.6.3/lib/pygments/popen.rb:381:in `rescue in get_header'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/pygments.rb-0.6.3/lib/pygments/popen.rb:362:in `get_header'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/pygments.rb-0.6.3/lib/pygments/popen.rb:259:in `block in mentos'
        /usr/lib/ruby/2.2.0/timeout.rb:88:in `block in timeout'
        /usr/lib/ruby/2.2.0/timeout.rb:32:in `block in catch'
        /usr/lib/ruby/2.2.0/timeout.rb:32:in `catch'
        /usr/lib/ruby/2.2.0/timeout.rb:32:in `catch'
        /usr/lib/ruby/2.2.0/timeout.rb:103:in `timeout'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/pygments.rb-0.6.3/lib/pygments/popen.rb:233:in `mentos'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/pygments.rb-0.6.3/lib/pygments/popen.rb:215:in `highlight'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/pygments.rb-0.6.3/lib/pygments/lexer.rb:140:in `highlight'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/github-linguist-3.5.2/lib/linguist/language.rb:519:in `colorize'
        /opt/chef-browser/lib/chef-browser/file_content.rb:41:in `highlight_file'
        /opt/chef-browser/lib/chef-browser/file_content.rb:30:in `show_file'
        /opt/chef-browser/lib/chef-browser/app.rb:203:in `block in <class:App>'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:1610:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:1610:in `block in compile!'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:974:in `[]'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:974:in `block (3 levels) in route!'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:993:in `route_eval'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:974:in `block (2 levels) in route!'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:1014:in `block in process_route'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:1012:in `catch'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:1012:in `process_route'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:972:in `block in route!'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:971:in `each'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:971:in `route!'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:1084:in `block in dispatch!'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:1066:in `block in invoke'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:1066:in `catch'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:1066:in `invoke'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:1081:in `dispatch!'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:906:in `block in call!'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:1066:in `block in invoke'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:1066:in `catch'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:1066:in `invoke'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:906:in `call!'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:894:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/session/abstract/id.rb:225:in `context'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/session/abstract/id.rb:220:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-protection-1.5.3/lib/rack/protection/xss_header.rb:18:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-protection-1.5.3/lib/rack/protection/path_traversal.rb:16:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-protection-1.5.3/lib/rack/protection/json_csrf.rb:18:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-protection-1.5.3/lib/rack/protection/base.rb:49:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-protection-1.5.3/lib/rack/protection/base.rb:49:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-protection-1.5.3/lib/rack/protection/frame_options.rb:31:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/nulllogger.rb:9:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/head.rb:13:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/show_exceptions.rb:21:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:181:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:2021:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/tempfile_reaper.rb:15:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/lint.rb:49:in `_call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/lint.rb:37:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/showexceptions.rb:24:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/commonlogger.rb:33:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/sinatra-1.4.6/lib/sinatra/base.rb:218:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/chunked.rb:54:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/rack-1.6.4/lib/rack/content_length.rb:15:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/puma-2.14.0/lib/puma/server.rb:541:in `handle_request'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/puma-2.14.0/lib/puma/server.rb:388:in `process_client'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/puma-2.14.0/lib/puma/server.rb:270:in `block in run'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/puma-2.14.0/lib/puma/thread_pool.rb:106:in `call'
        /opt/chef-browser/vendor/bundle/ruby/2.2.0/gems/puma-2.14.0/lib/puma/thread_pool.rb:106:in `block in spawn_thread'
10.0.2.2 - - [24/Nov/2015:14:33:44 +0000] "GET /cookbook/rcrs-base-0.1.25/recipes/certificates.rb HTTP/1.1" 500 183280 0.1477
```

Everything is running locally on my PC in Docker. These are the steps taken to create the containers:

```
docker run -d -t -i --name="chef.docker.test" --hostname="chef.docker.test" --privileged=true -p 80:80 -p 443:443 -e PUBLIC_URL="https://chef.docker.test" -e KNIFE_HOME="/etc/chef" quay.io/3ofcoins/chef-server "/opt/opscode/embedded/bin/ruby" "/init.rb"

docker exec chef.docker.test chef-server-ctl user-create joerg joerg herzinger joerg@fake.com secret > .chef/client.pem
docker exec chef.docker.test chef-server-ctl org-create testing Testing -a joerg

cp .chef/client.pem /tmp && chmod ugo+rw /tmp/client.pem # VBox mount has wrong permissions
docker run --name="chef-browser.docker.test" --link /chef.docker.test:chef.docker.test -d -p 9292:9292 -e TITLE="Chef Browser" -e CHEF_SERVER_URL="https://chef.docker.test:443/organizations/testing" -e CHEF_CLIENT_NAME="joerg" -e CHEF_CLIENT_KEY="/config/client.pem" -e INSECURE_SSL="true" -v /tmp/client.pem:/config/client.pem:rw -w /opt/chef-browser quay.io/3ofcoins/chef-browser ./bin/rackup -o 0.0.0.0 -p 9292 config.ru
```

Also viewing a cookbook file through knife works (knife cookbook show rcrs-base 0.1.25 recipes certificates.rb), so I don't think the chef server is the problem.

**Comment by xamebax** [comment]
Hm, it doesn't seem to happen when you don't put chef-browser in a container... Give me some time to try and reproduce this error and I will be more helpful. Thanks for the detailed dump.

**Comment by joerg** [comment]
Hi,

Is there any progress concerning this? We will be switching to your docker-chef-server container next week (updated to chef-server 12.4, pull request will be coming soon) and so I had a look if this error. Sadly, this also happens with the latest chef server version.

EDIT:
Good new, I found the error. Pygments (which you are using for file highlighting) request Python2, but Ubuntu comes by default with Python3. A simple `apt-get install python-virtualenv` fixed all problems. :smile: . Do you prefer a pull request or do you simple want to add this to the Dockerfile yourself?

**Comment by mpasternacki** [comment]
Fixed by #60
